### PR TITLE
[DistillationTrainer refactor] Constructor cleanup: safe internal cleanups

### PR DESCRIPTION
**Pull request by qgallouedec**
*Item 40a (Group F) — split from item 40, stacked on PR 39. Part of #6449.*

Item 40 is split because its signature-changing parts are coupled to `ServerDistillationTrainer` (its ctor forwards `model=None`/`compute_metrics`/`preprocess_logits_for_metrics` to the base `super().__init__`). This PR is the subset with **no API change and no server impact**:

- **Drop the dead inline `json.loads`** of `model_init_kwargs`/`teacher_model_init_kwargs` — `_VALID_DICT_FIELDS` already parses any JSON-string form in `DistillationConfig.__post_init__`, so they are dicts (or None) by the time the ctor runs. Switched to GRPO's copy-safe `dict(args.… or {})` so the subsequent `setdefault`s don't mutate the config object.
- **Guard the Liger build** with `is_liger_kernel_available()` (raises a clear `ImportError` — `pip install liger-kernel` — instead of a `NameError` when liger isn't installed), matching GRPO's ctor.
- **`processing_class.pad_token is None`** instead of `getattr(processing_class, "pad_token", None) is None` — a tokenizer always has the attribute.

Deferred to follow-ups: **40b** (drop `compute_metrics`/`preprocess_logits_for_metrics`, make `model` required — must update the server ctor in the same PR); **40c** (`quantization_config` + QLoRA cast, PEFT `enable_input_require_grads`/ZeRO-3 reentrant). imports consolidation → item 46.

Verified: ruff clean; full distillation + server suites 52 green.

---

> [!NOTE]
> **Low Risk**
> Constructor-only refactors with clearer errors and copy-safe kwargs; behavior is unchanged aside from a better Liger missing-dependency message.
> 
> **Overview**
> Internal **`DistillationTrainer.__init__`** cleanup with no public API change.
> 
> **Model init kwargs:** Removes redundant inline `json.loads` for `model_init_kwargs` / `teacher_model_init_kwargs` (already normalized in `DistillationConfig.__post_init__` via `_VALID_DICT_FIELDS`). Uses `dict(args.… or {})` copies before `setdefault` so the config object is not mutated.
> 
> **Liger:** When `use_liger_kernel` is enabled, checks `is_liger_kernel_available()` and raises a clear `ImportError` (`pip install liger-kernel`) instead of failing later with `NameError` when the optional import was skipped.
> 
> **Tokenizer:** Replaces `getattr(processing_class, "pad_token", None) is None` with `processing_class.pad_token is None`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 1aa84f9af58aa6fed676ada78928f42726c5e4f4. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/experimental/distillation/distillation_trainer.py [hunk 1]
```diff
@@ -347,16 +347,10 @@ def __init__(
...
-            model_init_kwargs = json.loads(model_init_kwargs)
-        teacher_model_init_kwargs = args.teacher_model_init_kwargs or {}
-        if isinstance(teacher_model_init_kwargs, str):
-            import json
-
-            teacher_model_init_kwargs = json.loads(teacher_model_init_kwargs)
+        # `_VALID_DICT_FIELDS` already parses any JSON-string form of these in `DistillationConfig.__post_init__`, so
+        # they are dicts (or None) here; copy so the setdefaults below don't mutate the config.
+        model_init_kwargs = dict(args.model_init_kwargs or {})
+        teacher_model_init_kwargs = dict(args.teacher_model_init_kwargs or {})
```
[the remark this answers is not in the document]
Comment (qgallouedec) [reply trl/experimental/distillation/distillation_trainer.py:369]: The JSON-string parsing lives upstream in transformers' `TrainingArguments.__post_init__` (strict `startswith("{")`), and GRPO shares the exact same base config + `dict(args.model_init_kwargs or {})` line, so this isn't distillation-specific, and normalizing here would diverge from GRPO. The iw_opd copy is a frozen quarantine. Leaving as-is.

**Review by kashif — APPROVED** [review]
LGTM — safe constructor cleanups with no behaviour change beyond matching GRPO’s `model_init_kwargs` handling. The whitespace-JSON parsing edge case is correctly left to upstream TrainingArguments / shared GRPO path.
